### PR TITLE
Some changes to Cuboid6

### PR DESCRIPTION
**Pull request by nikita488**

### CODE REVIEW

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 1]
```diff
@@ -70,10 +70,14 @@ public Cuboid6 set(Vector3 min, Vector3 max) {
...
 
+    public Cuboid6 set(Vec3i min, Vec3i max) {
+        return set(min.getX(), min.getY(), min.getZ(), max.getX(), max.getY(), max.getZ());
+    }
+    
     public Cuboid6 set(Cuboid6 c) {
         return set(c.min.x, c.min.y, c.min.z, c.max.x, c.max.y, c.max.z);
     }
-
+
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java]: Trailing spaces.

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 2]
```diff
@@ -88,8 +92,8 @@ public Cuboid6 add(Vector3 vec) {
         return add(vec.x, vec.y, vec.z);
     }
 
-    public Cuboid6 add(BlockPos pos) {
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java]: This breaks binary compatibility.

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 3]
```diff
@@ -106,8 +110,8 @@ public Cuboid6 subtract(Vector3 vec) {
         return subtract(vec.x, vec.y, vec.z);
     }
 
-    public Cuboid6 subtract(BlockPos pos) {
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java]: This breaks binary compatibility.

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 4]
```diff
@@ -124,6 +128,23 @@ public Cuboid6 expand(Vector3 vec) {
         return expand(vec.x, vec.y, vec.z);
     }
 
+    public Cuboid6 expandSide(EnumFacing side, int amount) {
+        switch (side.getAxisDirection()) {
+            case NEGATIVE:
+                min.add(side.getDirectionVec().getX() * amount, side.getDirectionVec().getY() * amount, side.getDirectionVec().getZ() * amount);
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java]: `min.add(new Vector3(side.getDirectionVec()).multiply(amount));` ?

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 5]
```diff
@@ -176,54 +197,80 @@ public boolean intersects(Cuboid6 b) {
...
-        throw new IndexOutOfBoundsException("Switch Falloff");
+        return 0;
     }
 
-    public Cuboid6 setSide(int s, double d) {
-        switch (s) {
-            case 0:
+    @Deprecated
+    public double getSide(int s)
+    {
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java]: Brace shouldn't be on a new line.

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 6]
```diff
@@ -176,54 +197,80 @@ public boolean intersects(Cuboid6 b) {
...
                 break;
-            default:
-                throw new IndexOutOfBoundsException("Switch Falloff");
         }
         return this;
     }
 
+    @Deprecated
+    public Cuboid6 setSide(int s, double d)
+    {
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java]: Brace shouldn't be on a new line.

**Review by covers1624 — CHANGES_REQUESTED** [review]
Some minor changes.

File: src/main/java/codechicken/lib/vec/Cuboid6.java (hunk 2 above)
Comment (nikita488) [reply src/main/java/codechicken/lib/vec/Cuboid6.java]: Well, i need revert back this change? I changed this because BlockPos extends Vec3i

Comment (covers1624) [reply src/main/java/codechicken/lib/vec/Cuboid6.java]: The JVM doesn't care about class hierarchy when invoking a method all it knows is its calling method add(Lnet/minecraft/math/blah/BlockPos;) not add(Lnet/minecraft/math/blah/Vec3i;). Add the new method and overload with a BlockPos one, delegate the call in blockPos to the new one with Vec3i.

**Comment by nikita488** [comment]
Fixed

**Comment by covers1624** [comment]
Looks good now, its on you if things explode :D

**Comment by nikita488** [comment]
Well, if something breaks, i fix this :P

**Pull request merged**